### DISCUSSION TITLE
Export-PnPPowerApp - using Azure AD/Entra ID Enterprise App

### DISCUSSION THREAD

**Discussion by renato87milagres** [pnp/powershell#4297] [General]
### Discussed in https://github.com/pnp/powershell/discussions/3743

<div type='discussions-op-text'>

<sup>Originally posted by **pkbullock** February 10, 2024</sup>
Hi I am looking to see if the Export-PnPPowerApp, can run under Azure AD App, instead of -Interactive.

I've tried using but are continually getting access denied in the following:

-  a App running under App Permissions i.e. Non Interactive login
- Giving app Power App Service permissions in Azure AD Permissions blade
- Within the Power Platform Admin, adding the Azure AD app in the app list on the environment. 
- Also adding the app as a service principal in the PP apis.

I believe this is possible, to achieve but I feel I am missing something here, just wondering if anyone knows or has tried this.

https://pnp.github.io/powershell/cmdlets/Export-PnPPowerApp.html
</div>

**Comment by LucasCXavier** [comment]
Hi team,

I've been trying to run the Export-PnPPowerApp cmdlet for several hours but haven't had any success so far.
Even when authenticating via the .pfx certificate, I'm encountering a Forbidden error.

The error message says that the service principal doesn't have permission to access:

https://10.0.5.20:20273/providers/Microsoft.BusinessAppPlatform/environments/

The App Registration has already been added as System Administrator in the target environment, so it's unclear what's missing.

Has anyone faced something similar before or knows if there are additional permission steps required (perhaps at the Enterprise Application level or API permissions in Azure)?

Thanks,
Lucas

**Comment by DougPunchak** [comment]
I too have been getting this error for months now.  When using the old "UseWebLogin" it worked perfectly.  The command-let doesn't seem to have outlined what permissions are required in the AppRegistration, or anywhere else, to make this work.

Can someone\anyone help?
